[lordidiot] iP

lordidiot opened the pull request:
# A heading

* A
* bullet
* list

1. A
2. numbered
3. list

```python
def a_fenced_code_block():
    with syntax_highlighting() as x:
        pass
```

- [x] add a task list
- [ ] watch lecture

🤓 

> a blockquote

[a hyper link](https://github.com/nus-cs2103-AY2324S1/ip/pull/527)

`inline code?`

some text formatting: **bold**, *italic*, ~~strikethrough~~ etc.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

lordidiot committed:
rename bot and add skeletal conversation

lordidiot committed:
add echo and bye convo

lordidiot committed:
add task adding and listing

lordidiot committed:
add task marking

lordidiot committed:
add task types (todo, event, deadline)

lordidiot committed:
add e2e test

lordidiot committed:
implement input error handling

lordidiot committed:
add delete functionality

lordidiot committed:
make test script executable

lordidiot committed:
Add task store and load

lordidiot committed:
Ignore data files

lordidiot committed:
Merge branch 'branch-Level-7'

lordidiot committed:
Improve datetime handling

lordidiot committed:
Merge branch 'branch-Level-8'

lordidiot committed:
Refactor and introduce parser class

lordidiot committed:
Abstract UI class

lordidiot committed:
Abstract TaskList class

lordidiot committed:
Change serialization format

lordidiot committed:
Fix wildcard import

lordidiot committed:
Handle invalid commands

lordidiot committed:
Merge branch 'branch-moreoop'

lordidiot committed:
Merge remote-tracking branch 'origin/add-gradle-support'

lordidiot committed:
Package into duke package

lordidiot committed:
Fix application class

lordidiot committed:
Add JUnit tests and fix errors surfaced

lordidiot committed:
Add javadocs for public methods

lordidiot committed:
Add checkstyle and fix coding standard violations

lordidiot committed:
Add find command

lordidiot committed:
Merge branch 'branch-codingstandard'

lordidiot committed:
Merge branch 'branch-Level-9'

lordidiot committed:
Fix styling errors

lordidiot committed:
Add JavaFX and prototype GUI

lordidiot committed:
Update functionality to work with GUI

Respirayson commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,116 @@
+    private static boolean handleInput(Parser parser, TaskList taskList, Ui ui) throws DukeException {
+        Task task;
+        switch (parser.getCommand()) {
+        case "bye":
```
perhaps it would be better to consider the case that the user might input the correct word but a different capitalisation? For eg. instead of bye, the user inputs Bye or BYE.

lordidiot replied:
Definitely, will work on that.

Respirayson commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,116 @@
+        ui.exit();
+    }
+
+    private static boolean handleInput(Parser parser, TaskList taskList, Ui ui) throws DukeException {
```
I like the function in that it is succinct and it is very readable. However, I think you might want to consider refactoring the code and extracting this function into the Parser class as that class deals with making sense of the user commands.

Respirayson commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,114 @@
+     * Initialise UI. Prints banner.
+     */
+    public void init() {
+        System.out.println("____________________________________________________________");
```
Small thing but perhaps it would be better to turn the underline into a constant so it can be easily reused elsewhere

lordidiot replied:
Thank you! That makes sense, will do so.

Respirayson commented on src/main/java/duke/TaskStorage.java:
```diff
@@ -0,0 +1,107 @@
+                return new ArrayList<>();
+            }
+
+            ArrayList<Task> tasks = new ArrayList<>();
```
I like how you used tasks as the variable name for the Java Collections Object

Respirayson commented on src/test/java/duke/mocks/TaskMock.java:
```diff
@@ -0,0 +1,21 @@
+package duke.mocks;
+
+import duke.Task;
+public class TaskMock extends Task {
```
I really like the usage of stubs when performing your unit testing. Really shows you went the extra mile!

Respirayson commented on src/main/java/duke/TaskStorage.java:
```diff
@@ -0,0 +1,107 @@
+     * @throws DukeException Any errors encountered while storing
+     */
+    public void storeTasks(List<Task> tasks) throws DukeException {
+        String serialized = tasks.stream()
```
Very cool usage of streams 👍

Respirayson reviewed (commented):
Looks good 👍
Just a few nitpicks here and there

lordidiot committed:
Add assertion to input handling

lordidiot committed:
Refactor Ui class

Minimise code reptition by replacing the dotted
line string constants with a single static
reference to a class constant DOTTED_LINE.

lordidiot committed:
Merge pull request #1 from lordidiot/branch-A-Assertions

Add assertions

lordidiot committed:
Merge pull request #2 from lordidiot/branch-A-CodeQuality

Refactor to improve code quality

lordidiot committed:
Improve search flexbility

lordidiot committed:
Add gradle CI

lordidiot committed:
Refactor command handling

lordidiot committed:
Reorganise Ui hierarchy and package structure

lordidiot committed:
Improve error message

lordidiot committed:
Fix style errors

lordidiot committed:
Improve GUI spacing and sizing

lordidiot committed:
Enlarge image

lordidiot committed:
Add product image

lordidiot committed:
Update user guide

lordidiot committed:
Add screenshot to user guide

lordidiot committed:
Fix gradle cross-platform build
